lib/order.rb

```diff
@@ -1,18 +1,70 @@
+require 'csv'
+require 'pry'
+
 module Grocery
   class Order
     attr_reader :id, :products
 
+    @@all_orders = []
+
     def initialize(id, products)
       @id = id
       @products = products
+      @tax = 0.075
+    end
+
+    def self.all
+      if @@all_orders.length > 0
+        return @@all_orders
+      end
```

Comment:
I like this - if we've already loaded the orders once, why do it again.

This technique is called "memoization", and it's a form of cacheing. There's potentially a big performance benefit here, because reading from disk is extremely expensive, but the downside is we don't have any way to pick up changes to the file made by a user or another program.